tools/erc-repository-indexer/erc-contract-indexer/src/services/registryGenerator.ts

```diff
@@ -39,13 +39,44 @@ export class RegistryGenerator {
    */
   private readonly erc721JsonFilePath: string;
 
+  /**
+   * @private
+   * @readonly
+   * @property {string} nextPointerFilePath - The file path where the next pointer for indexing will be stored.
+   */
+  private readonly nextPointerFilePath: string;
+
   constructor() {
     this.erc20JsonFilePath = Helper.buildFilePath(
       constants.ERC_20_JSON_FILE_NAME
     );
     this.erc721JsonFilePath = Helper.buildFilePath(
       constants.ERC_721_JSON_FILE_NAME
     );
+    this.nextPointerFilePath = Helper.buildFilePath(
+      constants.GET_CONTRACTS_LIST_NEXT_POINTER_JSON_FILE_NAME
+    );
+  }
+
+  /**
+   * Updates the next pointer in a file if it is not null.
+   * @param {string | null} next - The next pointer to be written to the file. If null, the file will not be updated.
+   * @returns {Promise<void>} A promise that resolves when the next pointer has been successfully written to the file.
+   */
+  async updateNextPointer(next: string | null): Promise<void> {
+    if (next) {
```

Comment:
How often is this called?
Just curious how often we're writing to file

Reply:
I plan to write this at the end of each batch of contracts, which is currently set to a maximum of 100. So if there are 1,000 contracts, this will be executed 10 times.